src/rich-text/inputrules.ts

```diff
@@ -17,7 +17,7 @@ export const emphasisRegexAsterisk = /\*([^*\s]([^*])*[^*\s]|[^*\s])\*$/;
 // matches: __some text__, but not __ text __
 const strongRegexUnderscores = /__(\S(?:|.*?\S))__$/;
 // matches: _some text_, but not __text_ or _ text _
-const emphasisRegexUnderscore = /_([^_\s]([^_])*[^_\s]|[^_\s])_$/;
+const emphasisRegexUnderscore = /(?<!\S)_([^_\s]([^_])*[^_\s]|[^_\s])_$/;
```

Comment:
would we also need to apply this to the double-underscore version above?

right now, the same bug would appear with something like `text__no__match` - does that matter?

Reply:
We can do that. There might be cases where people name their DB tables with a double underscore. 😀
Honestly I was thinking that maybe we could apply this non-whitespace negative lookahead for all the rules but then I was like if somebody has a problem with how things work they will tell us. No need for premature optimization.

Reply:
Addressed by https://github.com/StackExchange/Stacks-Editor/pull/262/commits/31d00a72f862432dabb59240976d3d27ff579f5a